packages/thirdweb/src/transaction/transaction-store.ts

```diff
@@ -0,0 +1,37 @@
+import type { Address } from "../utils/address.js";
+import type { Hex } from "../utils/encoding/hex.js";
+
+type StoredTransaction = {
+  transactionHash: Hex;
+  chainId: number;
+};
+
+const transactionStore = new Map<Address, StoredTransaction[]>();
```

Comment:
can you set this up to work as a "external store" out of the box:
- see: https://react.dev/reference/react/useSyncExternalStore
- other things we already have using this (example): `pendingSwapTx.ts` -> `createStore()`

it should work just fine in bare typescript, and then when we want to add this to react we can simply use `useSyncExternalStore` to hook into it (instead of having to figure out our own way to hook into it)

Reply:
essentially you'll have to have a way to:
- get the current value of the store
- set the current value of the store (update)
- register listeners that get called when something updates the value

it should be relatively straight forward to make this work that way here